src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -153,71 +153,56 @@
      * Indicates that we are about to go to the next or previous entry
      */
     private final BooleanProperty movingToDifferentEntry = new SimpleBooleanProperty();
-    private final EntryType entryType;
+    private EntryType entryType;
     private SourceTab sourceTab;
+    private final BorderLayout layout;
+    private TypeLabel typeLabel;
 
-    public EntryEditor(JabRefFrame frame, BasePanel panel, BibEntry entry, String lastTabName) {
-        this.frame = frame;
+    public EntryEditor(BasePanel panel, BibEntry entry) {
+        this.frame = panel.frame();
         this.panel = panel;
-        this.entry = Objects.requireNonNull(entry);
-        entry.registerListener(this);
-        entryType = EntryTypes.getTypeOrDefault(entry.getType(),
-                this.frame.getCurrentBasePanel().getBibDatabaseContext().getMode());
-
-        displayedBibEntryType = entry.getType();
 
         writeXmp = new WriteXMPEntryEditorAction(panel, this);
 
-        BorderLayout borderLayout = new BorderLayout();
-        setLayout(borderLayout);
-        setupToolBar();
+        layout = new BorderLayout();
+        setLayout(layout);
 
         container = OS.LINUX ? new CustomJFXPanel() : new JFXPanel();
+        // Create type-label
+        typeLabel = new TypeLabel("");
+        setEntry(entry);
+        setupToolBar();
+        container.setScene(new Scene(tabbed));
```

Comment:
I'm not sure but I think `setScene` should be called in the JavaFX thread (`DefaultTaskExecutor.runInJavaFXThread`).

Reply:
Done